### PR TITLE
SoftDeletes support

### PR DESCRIPTION
**Pull request by royduin**

### CODE REVIEW

**Comment by freekmurze** [comment]
Thank you for this. 

Because I don't use them myself I don't want to maintain support for soft deletes. 

Do this in your own project.

**Comment by royduin** [comment]
SoftDeletes is core functionality of Laravel, there are more people requesting this: https://github.com/spatie/laravel-medialibrary/issues?utf8=%E2%9C%93&q=is%3Aissue%20soft, because you're not using it you won't accept it?

**Comment by freekmurze** [comment]
> because you're not using it you won't accept it

Yeah, in general I only maintain what I use myself.

Maybe I have been to harsh in my answer. If you PR some tests that makes sure this works, I might reconsider my opinion.

**Comment by royduin** [comment]
I'll create some tests tomorrow 👍

**Comment by royduin** [comment]
@freekmurze, can you reopen the PR? I've created the tests

**Comment by freekmurze** [comment]
Please also make sure that the `clean` command plays nice with `SoftDeletes` as well. Maybe add some tests there too.

**Comment by royduin** [comment]
That's working fine because the `medialibrary:clean` command just matches directories with the media table. It's not checking the `model_id`.

**Comment by royduin** [comment]
Any idea why the test is failing? When all tests ran successful the PR will be merged and a new release will be created?

**Comment by freekmurze** [comment]
I'll take care of this when I have some time.

File: src/HasMedia/HasMediaTrait.php [hunk 1]
```diff
@@ -36,6 +36,12 @@ public static function bootHasMediaTrait()
                 return;
             }
 
+            if (in_array('Illuminate\Database\Eloquent\SoftDeletes', class_uses(__CLASS__))) {
```
Comment (freekmurze) [inline src/HasMedia/HasMediaTrait.php]: Use `trait_uses_recursive` here and use the `::class` constant.

File: tests/Media/DeleteTest.php [hunk 2]
```diff
@@ -47,7 +79,7 @@ public function shouldDeletePreservingMedia()
     /**
      * @test
      */
-    public function it_will_remove_the_files_when_shouldDeletePreservingMedia_returns_true()
+    public function it_will_remove_the_files_when_shouldDeletePreservingMedia_returns_false()
```
Comment (freekmurze) [inline tests/Media/DeleteTest.php]: Nice catch, thanks!

File: src/HasMedia/HasMediaTrait.php (hunk 1 above)
Comment (royduin) [inline src/HasMedia/HasMediaTrait.php]: Fixed

**Comment by royduin** [comment]
Fixed and I've created a PR for the docs: https://github.com/spatie/docs.spatie.be/pull/180

**Comment by freekmurze** [comment]
Thank you very much for your work on this.

**Comment by royduin** [comment]
Thanks for merging!

**Pull request merged**